alexa/skill/news.php

```diff
@@ -0,0 +1,230 @@
+					);
+
+					$result = $this->endpoint_content( $args );
+
+					$response
+						->respond( $result['content'] )
+						/* translators: %s: site title */
+						->withCard( sprintf( __( 'Latest on %s', 'alexawp' ), get_bloginfo( 'name' ) ) )
+						->addSessionAttribute( 'post_ids', $result['ids'] );
+					break;
+				case 'ReadPost':
+					if ( $post_number = $request->getSlot( 'PostNumberWord' ) ) {
+						if ( 'second' === $post_number ) {
+							/**
+							* Alexa Skills Kit passes 'second' instead of '2nd'
+							* unlike the case for all other numbers.
+							*/
+							$post_number = 2;
+						} else {
+							$post_number = substr( $post_number, 0, -2 );
+						}
+					} else {
+						$post_number = $request->getSlot( 'PostNumber' );
+					}
+
+					if ( ! empty( $request->session->attributes['post_ids'] ) && ! empty( $post_number ) ) {
+						$post_id = $this->get_post_id( $request->session->attributes['post_ids'], $post_number );
+						$result = $this->endpoint_single_post( $post_id );
+						$response
+							->respond( $result['content'] )
+							->withCard( $result['title'], '', $result['image'] )
+							->endSession();
+					} else {
+						$this->message( $response );
+					}
+					break;
+				case 'AMAZON.StopIntent':
+				case 'AMAZON.CancelIntent':
+					$this->message( $response, 'stop_intent' );
+					break;
+				case 'AMAZON.HelpIntent':
+					$this->message( $response, 'help_intent' );
+					break;
+				default:
+					$this->skill_intent( $intent, $request, $response );
+					break;
+			}
+		} elseif ( $request instanceof \Alexa\Request\LaunchRequest ) {
+			$this->message( $response, 'launch_request' );
+		}
+	}
+
+	/**
+	 * Handles intents that come from outside the main set of News skill intents
+	 * @param string $intent name of the intent to handle
+	 * @param AlexaRequest $request
+	 * @param AlexaResponse $response
+	 */
+	private function skill_intent( $intent, $request, $response ) {
+		$custom_skill_index = get_option( 'alexawp_skill_index_map', array() );
+		if ( isset( $custom_skill_index[ $intent ] ) ) {
+			$alexawp = Alexawp::get_instance();
+			$alexawp->skill_dispatch( absint( $custom_skill_index[ $intent ] ), $request, $response );
+		}
+	}
+
+	/**
+	 * Packages up the post data that will be served in the response
+	 * @param int $id ID of post to get data for
+	 * @return array Data from the post being returned
+	 */
+	private function endpoint_single_post( $id ) {
+		$transient_key = 'voicewp_single_' . $id;
+		if ( false === ( $result = get_transient( $transient_key ) ) ) {
+			$single_post = get_post( $id );
+			$post_content = preg_replace( '|^(\s*)(https?://[^\s<>"]+)(\s*)$|im', '', strip_tags( strip_shortcodes( $single_post->post_content ) ) );
+			$result = array(
+				'content' => $post_content,
+				'title' => $single_post->post_title,
+				'image' => get_post_thumbnail_id( $id ),
+			);
+			// TODO: If content of the post changes,
```

Comment:
I wouldn't call this a todo. I'd say required for launch. This could cause immediate problems on a site like People or EW if a story is republished due to a factual correction.